compiler/ast.nim

```diff
@@ -1054,6 +1054,20 @@ const
   defaultOffset = -1
 
 
+var emptyString = ""
```

Comment:
Come on... Instead try something like:

```nim

template stringProp(a: PNode; prop: untyped): bool =
  case a.kind
  of nkStrLit..nkTripleStrLit: prop a.strVal
  of nkSym: prop a.sym.name.s
  of nkIdent: prop a.ident.s
  else: false
```

Reply:
I've tried (see option 2 and option 3, a variant of your proposal), it's objectively worse:
* option 2: causes a copy, which is what i wanted to avoid in first place
* option 3: causes code bloat, like multiple yield statements in inline iterators

and both options lead to more complex client code. How would you translate code like this with option 2 or 3?
```nim
echo (n[1].getStrVal, n[2].getStrVal, n.kind)
```

## option 2: your suggestion
```nim
template strValProp*(a: PNode; prop: untyped): bool =
  let a2 = a # prevent double evaluation bugs
  case a2.kind
  of nkStrLit..nkTripleStrLit: prop a2.strVal
  of nkSym: prop a2.sym.name.s
  of nkIdent: prop a2.ident.s
  else: false
```

```nim
n[1].getStrVal.startsWith('\'')
```
gets replaced by:
```nim
from std/sugar import `=>`
n[1].strValProp((it: string) => it.startsWith('\''))
```

## option 3: your suggestion modified
```nim
template strValIt*(a: PNode; op: untyped): bool =
  let a2 = a # prevent double evaluation bugs
  case a2.kind
  of nkStrLit..nkTripleStrLit:
    template it: untyped {.inject.} = a2.strVal
    op
  of nkSym:
    template it: untyped {.inject.} = a2.sym.name.s
    op
  of nkIdent:
    template it: untyped {.inject.} = a2.ident.s
    op
  else: false
```

```nim
n[1].getStrVal.startsWith('\'')
```
gets replaced by:
```nim
n[1].strValIt(it.startsWith('\''))
```
which isn't too bad, but see there's no simple equivalent in other cases, eg:
```
echo (n[1].getStrVal, n[2].getStrVal, n.kind)
```